libdrgn/python/program.c

```diff
@@ -734,6 +734,66 @@ static StackTrace *Program_stack_trace(Program *self, PyObject *args,
 	return ret;
 }
 
+static PyObject *Program_symbols(Program *self, PyObject *args, PyObject *kwargs)
+{
+	static char *keywords[] = {
+		"name", NULL,
```

Comment:
Since this can be an address or name, this should probably be a positional-only argument like for `Program.symbol()`. So, change this to a `METH_O` method and document in `_drgn.pyi` that `address_or_name` is positional-only (we can't use the [PEP 570](https://www.python.org/dev/peps/pep-0570/) syntax yet since we still support Python 3.6 and 3.7).

Reply:
I chose `METH_VARARGS | METH_KWARGS` because `METH_O` does not allow for the caller to omit the argument:

```
TypeError: Program.symbols() takes exactly one argument (0 given)
```

I can however still make it positional-only by using `METH_VARARGS`, and it would probably be better since the kwargs are pointless as it is. (and the keyword argument `name` clashes with the documented parameter name in `_drgn.pyi`). So I think I'll end up switching to `METH_VARARGS`.

Reply:
Good point, `METH_VARARGS` makes more sense here.